### ISSUE TITLE
Instability when testing docker-default profile

### ISSUE THREAD

**Issue by whyseu** [moby/moby#43396]
**Steps to reproduce the issue:**
1.We want to reproduce the CVE-2019-5736 vulnerability
2.We use the POC( https://github.com/Frichetten/CVE-2019-5736-PoC/blob/master/main.go)
3.We run the POC in some public cloud server(use the unmodified apparmor docker-default profile)

**Describe the results you received:**
We run the POC about 100 times. 
(1) Most of the result is “attack is failure and the apparmor don't have any log”.
(2) About 5 attacks were successful and apparmor are **not logged**
(3) About 5 attacks were successful and apparmor are **logged**
![image](https://user-images.githubusercontent.com/35478590/159109144-6e334101-d819-42c2-9508-c61c6ca67d78.png)

**Describe the results you expected:**
 We are confused. We use the same POC and same test enviroment. There shouldn't be so many different situations.  
(1) If the attack is completed, the apparmor should be deterministically logged or not logged
(2)The docker-default profile should deterministically block the N-day CVE POC

**Additional information you deem important (e.g. issue happens only occasionally):**
Written above in the "Describe the results you received" section.

**Output of `docker version`:**

```
Client:
 Version:      18.03.1-ce
 API version:  1.37
 Go version:   go1.9.5
 Git commit:   9ee9f40
 Built:        Wed Jun 20 21:43:51 2018
 OS/Arch:      linux/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.03.1-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.9.5
  Git commit:   9ee9f40
  Built:        Wed Jun 20 21:42:00 2018
  OS/Arch:      linux/amd64
  Experimental: false
```

**Output of `docker info`:**

```
Containers: 1
 Running: 1
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 18.03.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 773c489c9c1b21a6d78b5c538cd395416ec50f88
runc version: 4fc53a81fb7c994640722ac585fa9ca548971871
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.15.0-159-generic
Operating System: Ubuntu 18.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.79GiB
Name: VM-4-6-ubuntu
ID: J7GJ:PCA4:NQ5C:O6E7:3UI5:7M7Q:CF57:QR73:EZFX:JV63:GZWY:D4DH
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by thaJeztah** [comment]
I'm not sure I fully understand your question; you're trying to reproduce [CVE-2019-5736](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-5736), which was a security issue in runc (addressed in [runc v1.0.0-rc7](https://github.com/opencontainers/runc/releases/v1.0.0-rc7), included in docker 18.09.2 and up).

- You're running a version of docker (more notably, runc) that does not have the fix
- To my knowledge, AppArmor was not able to prevent the vulnerability (at least, no changes were made to the profile related to this CVE)

(Of course if improvements can be made to the AppArmor profile that's used, contributions are welcome)

**Comment by thaJeztah** [comment]
Let me close this ticket for now, as it looks like it went stale.
